MaxText/checkpointing.py

```diff
@@ -118,8 +119,11 @@ def map_to_pspec(data, pspec):
     else:
       step = load_from_other_directory_step
       max_logging.log(f"restoring state from {load_from_other_directory} step {step}")
-    return mngr_loader.restore(step, abstract_unboxed_pre_state,
-                                      {"restore_args" : restore_args}), None
+    state = mngr_loader.restore(step, abstract_unboxed_pre_state,
+                                      {"restore_args" : restore_args})
+
+    max_utils.convert_state_to_jnp(state.params)
```

Comment:
Why did you feel a need to do this? If we're going to do this (which is arguably a good thing!), there is some other simplifications we should be making

Reply:
If we make the weights into jax.numpy, the checkpoint converter code (MaxText/convert_llama_ckpt.py ) will not run on a v4-8 (for decoding we are using tensor parallelism to make it work on a v4-8)

Reference saxml converter:
https://github.com/google/saxml/blob/2e59631958d03b8f0858f33da185a96c7a9ddb1d/saxml/tools/convert_llama_ckpt.py

Reply:
What isn't jax.numpy.array? Everything we serialize really should be a jax.numpy.array already...

Reply:
converted to jax.numpy.array